Gopkg.toml

```diff
@@ -34,285 +34,489 @@ required = [
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/errors"
   revision = "fc3f73a224499b047eda7191e5d22e1e9631e86f"
 
+  # main-usage = "api/v1"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/loads"
   revision = "bf98340e04734f11563fd3008e87be4c1d4ffc08"
 
+  # main-usage = "swagger spec"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/runtime"
   revision = "bf2ff8f7150788b1c7256abb0805ba0410cbbabb"
 
+  # main-usage = "api/v1"
+  # on-revision = ""
+
 # Once we update "github.com/go-openapi/runtime" we can remove this override
 [[override]]
   name = "github.com/go-openapi/analysis"
   revision = "d5a75b7d751ca3f11ad5d93cfe97405f2c3f6a47"
 
+  # main-usage = "api/v1"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/spec"
   revision = "3a0434164aa36744c7ba29b822d36894a1e5ec96"
 
+  # main-usage = "api/v1"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/strfmt"
   revision = "0cb3db44c13bad3b3f567b762a66751972a310cc"
 
+  # main-usage = "swagger and pkg/controllers"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/swag"
   revision = "96d7b9ebd181a1735a1c9ac87914f2b32fbf56c9"
 
+  # main-usage = "api/v1"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/go-openapi/validate"
   revision = "b11dbf889b64cf815a7e2667412df313f4a3f17b"
 
+  # main-usage = "api/v1"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/gorilla/mux"
   revision = "26a6070f849969ba72b72256e9f14cf519751690"
 
+  # main-usage = "plugins/cilium-docker"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/hashicorp/consul"
   revision = "05bc9149a5204b213a61bc410b8cec21f10a23fe"
 
+  # main-usage = "pkg/kvstore"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/hashicorp/go-immutable-radix"
-  version = "1.0.0"
+  revision = "v1.0.0"
+
+  # main-usage = "cilium/cmd"
+  # on-revision = ""
 
 [[constraint]]
   name = "github.com/hashicorp/go-version"
   revision = "03c5bf6be031b6dd45afec16b1cf94fc8938bc77"
 
+  # main-usage = "multiple packages"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/jessevdk/go-flags"
   revision = "4e64e4a4e2552194cf594243e23aa9baf3b4297e"
 
+  # main-usage = "all CLI packages"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/kevinburke/ssh_config"
   revision = "db49ba357de1f26c56dac48a5de39c65785bf24a"
 
+  # main-usage = "only in integration tests"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/kr/pretty"
   revision = "cfb55aafdaf3ec08f0db22699ab822c50091b1c4"
 
+  # main-usage = "pkg/comparator"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/mattn/go-shellwords"
   revision = "9858af9cca4c73576f0b8c6609a396eb0878023c"
 
+  # main-usage = "pkg/datapath/iptables"
+  # on-revision = ""
+
 [[constraint]]
   name = "github.com/miekg/dns"
-  version = "1.0.8"
+  revision = "v1.0.10"
```

Comment:
This dependency change seems out of place in this commit?

Reply:
[1] No, it was not a mistake. The only reason I prefer `revision` over `version`, and `revision` over `branch` is so that we have the same version whenever a developer do `dep ensure`. The `version` tag specifies the minimal version that Cilium requires. In this particular case, the `version` was '1.0.8' but in reality the one that we were using to compile Cilium was `1.0.10`, check https://github.com/cilium/cilium/pull/6301/files/6b77461ad27a7a5e33ba2975115773629ba27d6b..e738f84977e3b4dbb000980c87a75079ccd09649#diff-836546cc53507f6b2d581088903b1785

I personally think it's better to have a revision so we can be clearly explicit what library revisions we are using.

Reply:
Understood, thanks for the explanation. That seemed like a dependency change in a commit that otherwise just added comments.